### ISSUE TITLE
Vulnerability Detector - Mismatch in generated CPEs for Windows Server

### ISSUE THREAD

**Issue by MarcelKemp** [wazuh/wazuh#24534]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0 | Vulnerability Detection | Manager | All | Windows Server |

## Description

I have detected a problem when detecting vulnerabilities in all Windows Server agents, causing all vulnerabilities in the system not to appear (generating false negatives in all agents). 

The problem is due to the CPEs we generate and use to match the NVD CPEs. 

### Technical analysis

Currently, the CPEs we generate have the following format:
- `microsoft:windows_server_2016:$(RELEASE):::::`
  - Substitute `$(RELEASE)` for the values `1803`, `1903`, `1909`, etc. E.g.: `microsoft:windows_server_2016:1903:::::`
  - Which, as we can see in the following link, is deprecated: https://nvd.nist.gov/products/cpe/detail/10408E3C-0F82-4B96-8C0E-529C2BF3B2B4?namingFormat=2.3&orderBy=CPEURI&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_server_2016%3A%3A%3A%3A%3A%3A%3A%3A&status=FINAL%2CDEPRECATED

Therefore, it is necessary to modify the CPE we generated so that it complies with the following format:
- `microsoft:windows_server_2016:$(VERSION):::::`
  - Where `$(VERSION)` is the full version shown by the system: `10.0.14393.5717`, `10.0.14393.6614`, etc. E.g.: `microsoft:windows_server_2016:10.0.14393.6614:::::`
  - Which, as you can see in the link, is the new format: https://nvd.nist.gov/products/cpe/detail/433809E1-2F8B-49FE-A6A0-E5F1FFABD1FE?namingFormat=2.3&orderBy=CPEURI&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_server_2016%3A%3A%3A%3A%3A%3A%3A%3A&status=FINAL

This change must be made at least for the following cases:
- WS2012 (except R2): https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_server_2012%3A%3A%3A%3A%3A%3A%3A%3A
- WS2016: https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_server_2016%3A%3A%3A%3A%3A%3A%3A%3A
- WS2019: https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_server_2019%3A%3A%3A%3A%3A%3A%3A%3A
- WS2022: https://nvd.nist.gov/products/cpe/search/results?namingFormat=2.3&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_server_2022%3A%3A%3A%3A%3A%3A%3A%3A

And check if there might be any extra cases.

On the other hand, a typo has been found in the case of _Windows Server 2022 (23H2)_, which needs to be fixed as well.

**Comment by Dwordcito** [comment]
@sebasfalcone please add RCA and create development issues.

**Comment by sebasfalcone** [comment]
@GabrielEValenzuela Also we need to fix [the typo](https://github.com/wazuh/intelligence-data/blob/9856d1cc42b00a2c42c10bba6f12924fecf5aac3/base-rules/oscpe-global.json#L26) on the windows server 2022 23H2 key

**Comment by GabrielEValenzuela** [comment]
# Update 

- Installed the following agents
     -  XP
     - Seven
     - 8.1
     - 10
     - 11
     - Server2016
     - Server2012R2

- Pending Server2019 and Server2022. Validating info and research through testcase.

**Comment by sebasfalcone** [comment]
@GabrielEValenzuela Take a look at this table:
- https://github.com/wazuh/intelligence-platform/issues/1583#issuecomment-2205464316

It a good analysis of what the content brings to us in the windows subsystem

**Comment by sebasfalcone** [comment]
Look at [Windows 10 CPES](https://nvd.nist.gov/products/cpe/detail/B7BEC64C-4B40-4C86-8FC0-ECD927813782?namingFormat=2.3&orderBy=CPEURI&keyword=cpe%3A2.3%3Ao%3Amicrosoft%3Awindows_10%3A21h1%3A*%3A*%3A*%3A*%3A*%3Ax86%3A*&status=FINAL%2CDEPRECATED), they were deprecated and converted from this:
- `cpe:2.3:o:microsoft:windows_10:21h1:*:*:*:*:*:x86:*`

Into this
- `cpe:2.3:o:microsoft:windows_10_21h1:-:*:*:*:*:*:x86:*`

**Comment by GabrielEValenzuela** [comment]
# Report analysis 

## Introduction
Effective test development and accurate platform enumeration are crucial for maintaining security and compatibility in widely used operating systems like Windows. This report addresses advancements and challenges associated with these tasks, focusing on improvements proposed in the GitHub pull request and the need to adapt our approach to deprecated CPEs.

## Methodology
Description of the process used for generating a test case `24` on efficacy test QA for different Windows versions.

## Results
Presentation of the results obtained from the test case generation, we have on this table the `os_name`, `os_version`, and the currently generated `CPE`.

|os_name                         |os_version      |             CPE Name                                    |
|--------------------------------|----------------|---------------------------------------------------------|
|Microsoft Windows XP            | 5.1.2600       | cpe:/o:microsoft:windows_xp::::::                       |
|Microsoft Windows Vista         | 6.0.6002       | cpe:/o:microsoft:windows_vista::::::                    |
|Microsoft Windows 7 Professional| 6.1.7601       | cpe:/o:microsoft:windows_7::::::                        |
|Microsoft Windows 8.1 Pro       | 6.3.9600       | cpe:/o:microsoft:windows_8.1::::::                      |
|Microsoft Windows 10 Pro        | 10.0.xxxx.2965 | cpe:/o:microsoft:windows_10_yyhz:10.0.xxxx.2965:::::    |
|Microsoft Windows 11 Enterprise | 10.0.xxxx.3296 | cpe:/o:microsoft:windows_11_yyhz:10.0.xxxx.3296:::::    |
|Microsoft Windows Server 2012 R2| 6.3.9600.20371 | cpe:/o:microsoft:windows_server_2012:r2:::::            |
|Microsoft Windows Server 2016   | 10.0.14393.2791| cpe:/o:microsoft:windows_server_2016::::::              |
|Microsoft Windows Server 2019   | 10.0.17763.1935| cpe:/o:microsoft:windows_server_2019::::::              |
|Microsoft Windows Server 2022   | 10.0.20348.2402| cpe:/o:microsoft:windows_server_2022::::::              |

> [!NOTE]  
> The `xxxx` and `yyHz` values differ from each release of Windows.

## Conclusion
In conclusion, it is imperative that we first enhance the content by replacing all available fields to potentially reduce and rectify false positives and false negatives. Consideration must be given to the changes implemented in [PR #1628](https://github.com/wazuh/intelligence-platform/pull/1628), as these modifications may necessitate a minor refactoring of the scanner code.

This approach aligns with the observations noted in the comment about invalid Windows versions under NVD - [intelligence-platform#1583 (comment)](https://github.com/wazuh/intelligence-platform/issues/1583#issuecomment). The proactive update of our testing and enumeration practices will greatly contribute to our ongoing efforts in maintaining robust and accurate system assessments.

**Comment by sebasfalcone** [comment]
## Issue blocked
We have a meeting to discuss this with @Damian-Mangold and @Dwordcito

**Comment by sebasfalcone** [comment]
## Conclusion
@Dwordcito Given that the solution to this problem is closely related to the results obtained here:
- https://github.com/wazuh/intelligence-platform/issues/1583

I think we should close this issue and pursue the update of the maps there (to prevent the back and forth between teams)

**Comment by sebasfalcone** [comment]
## Update ETA
Updated ETA so the final reviewer can see this issue

**Comment by Dwordcito** [comment]
This issue cannot be closed until the change is productive, and must remain blocked, since here we give the news/expectations of the resolution of this problem to the user who created #24242.

**Comment by c3rberus** [comment]
I reported the original issue in #24242 and was looking to see if there are any updates on this? There are a few links referenced in this thread, but intelligence-platform is not a repository I have access to (must be something private).

**Comment by sebasfalcone** [comment]
@c3rberus Yes, intelligence-platform is a private repository. There we maintain the generated vulnerability content

@Damian-Mangold The issue that will solve this issue is this one:
- https://github.com/wazuh/intelligence-platform/issues/1583

It was already merged, changes were deployed to production?

**Comment by Damian-Mangold** [comment]
> @c3rberus Yes, intelligence-platform is a private repository. There we maintain the generated vulnerability content
> 
> @Damian-Mangold The associated issue to this one was:
> 
> * [NVD - Windows invalid version  intelligence-platform#1583](https://github.com/wazuh/intelligence-platform/issues/1583)
> 
> It was already merged, were changes deployed to production?

The fix has been done. It will be in production in the next minor version of VDP.

**Comment by sebasfalcone** [comment]
## ETA updated
ETA updated to match next minor version of VDP

**Comment by sebasfalcone** [comment]
## Update
No work to be done

**Comment by GabrielEValenzuela** [comment]
Hi @Damian-Mangold ! 

Do you know if this is ready to continue? Due the minor of VDP

**Comment by Damian-Mangold** [comment]
> Hi @Damian-Mangold !
> 
> Do you know if this is ready to continue? Due the minor of VDP

Hi @GabrielEValenzuela, the deployment from last week was a patch version of VDP, which did not include the changes related to version translations. The version translations will be incorporated in VDP v1.1.0. For now, only translations for `windows_server_2008` and `windows_server_2012` have been added. Below is an example of what the content will look like:

```json
        {
          "cpes": [
            "cpe:2.3:o:microsoft:windows_server_2008:-:*:*:*:*:*:*:*",
            "cpe:2.3:o:microsoft:windows_server_2008:-:*:*:*:*:*:itanium:*",
            "cpe:2.3:o:microsoft:windows_server_2008:-:*:*:*:*:*:x64:*"
          ],
          "defaultStatus": "unaffected",
          "product": "windows_server_2008",
          "vendor": "microsoft",
          "versions": [
            {
              "status": "affected",
              "version": "6.0.6001"
            }
          ]
        }
```

```json
        {
          "cpes": [
            "cpe:2.3:o:microsoft:windows_server_2012:-:*:*:*:*:*:*:*"
          ],
          "defaultStatus": "unaffected",
          "product": "windows_server_2012",
          "vendor": "microsoft",
          "versions": [
            {
              "status": "affected",
              "version": "6.2.9200"
            }
          ]
        }
```

Please note that this functionality is currently limited to correcting versions only. The feature that addresses the correction of CPE content and format will be introduced in VDP v2.0.0.

**Comment by GabrielEValenzuela** [comment]
Thanks  Dami!

**Comment by sebasfalcone** [comment]
## Update
No changes on the ETA

**Comment by sebasfalcone** [comment]
## Update
No changes on the ETA

**Comment by sebasfalcone** [comment]
## Update
No changes on the ETA

**Comment by sebasfalcone** [comment]
## Update
No changes on the ETA

**Comment by sebasfalcone** [comment]
## Update
No changes on the ETA

**Comment by Dwordcito** [comment]
We can use this issue to test if the feed of 1.3.0 have the expected efficacy.

**Comment by Damian-Mangold** [comment]
Check if it is correct that the cpe name contains the release.

- [cpe:2.3:o:microsoft:windows_server_2012_r2:6.3.9600](https://github.com/wazuh/intelligence-data/blob/0e2947e542f9c643c2dc4d91876077212dd03895/vulnerabilities/intelligence-platform/translation-mapping.json#L185C137-L185C188)
- [cpe:2.3:o:microsoft:windows_server_2008_r2:6.1.7600](https://github.com/wazuh/intelligence-data/blob/0e2947e542f9c643c2dc4d91876077212dd03895/vulnerabilities/intelligence-platform/translation-mapping.json#L181C161-L181C212)

**Comment by Damian-Mangold** [comment]
This issue will be completed as a result of the realization of the following VDP objective:
 - [Expand CPE Translation Map](https://github.com/wazuh/vulnerability-detector-provider/issues/2039#top)
